src/pls.jl

```diff
@@ -217,10 +225,35 @@ function fitted!(v::AbstractArray{T}, m::LinearMixedModel{T}) where T
     v
 end
 
+fitted!(v::AbstractArray{T}, m::LinearMixedModel{T}) where {T} = fitted!(v, m, m.trms)
+
 StatsBase.fitted(m::LinearMixedModel{T}) where {T} = fitted!(Vector{T}(nobs(m)), m)
 
 StatsBase.predict(m::LinearMixedModel) = fitted(m)
 
+function StatsBase.predict(m::LinearMixedModel{T}, test::AbstractDataFrame; #, train::AbstractDataFrame;
+	 contrasts=Dict()) where T
+    mf = ModelFrame(m.formula, test, contrasts=contrasts)
+    X = ModelMatrix(mf).m
+    n = size(X, 1)
+    T = eltype(X)
+
+    # # can the tdict be used from the original data? (without building this modelframe again)
+    # mf_train = ModelFrame(m.formula, train, contrasts=contrasts)    
+    # tdict = get_tdict(mf_train.terms.terms)
+    tdict = get_tdict(mf.terms.terms)
+
+    # make a new trms object
+    trms = get_trms(tdict,mf.df,test,T,n)
+    
+    sort!(trms, by = nrandomeff, rev = true)
+    push!(trms, MatrixTerm(X, coefnames(mf)))
+    # for the indexing on fitted
+    push!(trms, MatrixTerm(X, coefnames(mf)))
```

Comment:
Why `X` twice?  I realize you don't necessarily have a `y` column in the data frame but you just need a column of zeros of type `T`.  Even if you do decide to push `X` onto the vector twice you can define and evaluate it once then push it twice.

Reply:
Yeah I had put `[]` but it wasn't happy with that. I think I'd just need `Array{T}`.